### PR TITLE
VM: Create bind mount of config drive and instance disk devices for exporting via 9p and virtiofsd

### PR DESCRIPTION
**Pull request by tomponline**
This works around the lack of read-only support in virtio-fs and provides an additional layer of protection for 9p shares.

### CODE REVIEW

**Comment by tomponline** [comment]
@stgraber ready for review

File: lxd/instance/drivers/driver_qemu.go [hunk 1]
```diff
@@ -529,6 +529,32 @@ func (d *qemu) Freeze() error {
 	return nil
 }
 
+// configDriveMountPath returns the path for the config drive bind mount.
+func (d *qemu) configDriveMountPath() string {
+	return filepath.Join(d.DevicesPath(), "config")
```
Comment (stgraber) [inline lxd/instance/drivers/driver_qemu.go:534]: What happens if I happen to define a device called `config` that's also a disk but a different one?

Comment (stgraber) [reply lxd/instance/drivers/driver_qemu.go:534]: I suspect the safe thing to do here is to mount it as a sub-directory of the instance mountpoint instead (same location where the writable config path is). So something like `config.mount` alongside the writable `config` directory.

**Comment by tomponline** [comment]
@stgraber marking this as ready for review now as it passes the local jenkins tests for dir and zfs (it needed a couple of fixes to ordering when the virtiofsd process was started and stopped for the config drive so as to not prevent root volume unmount). It may even help reduce the frequency that the restart unmount errors occur.

**Pull request merged**